llvm/lib/ProfileData/SampleProfReader.cpp

```diff
@@ -1788,8 +1788,11 @@ void SampleProfileReaderItaniumRemapper::applyRemapping(LLVMContext &Ctx) {
 
 std::optional<StringRef>
 SampleProfileReaderItaniumRemapper::lookUpNameInProfile(StringRef Fname) {
-  if (auto Key = Remappings->lookup(Fname))
-    return NameMap.lookup(Key);
+  if (auto Key = Remappings->lookup(Fname)) {
+    StringRef Result = NameMap.lookup(Key);
+    if (!Result.empty())
+      return Result;
```

Comment:
What about adding comments to explain when Remappers returns a valid key but NameMap returns an empty string?

Reply:
The comment in the header file should be clear enough that it returns a remapped name Only if a remapped symbol is found. The problem is the function's API design  is bad (mixing both empty string and null to represent non-existent value). This should be addressed somewhere else.